### ISSUE TITLE
Iterable argument doesn't accept set intersection

### ISSUE THREAD

**Issue by squaresmile** [microsoft/pyright#4576]
## Environment data

-   Language Server version: v2023.2.11
-   OS and version: Windows 11
-   Python version (& distribution if applicable, e.g. Anaconda): 3.11 python.org

## Code Snippet

```python
from typing import Iterable

a = {123, 456} 
b = {234, 789}
c = a | b

def func(iterable: Iterable[int | str]):
    print(iterable)
    
func(a | b)
func(c)
```

## Expected behavior

No type error

## Actual behavior

`func(a | b)` has the following error
```
Operator "|" not supported for types "set[int]" and "set[int]"
  Operator "|" not supported for types "set[int]" and "set[int]" when expected type is "Iterable[int | str]"
```
`func(c)` has no error

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.293, which I just published. It will also be included in a future release of pylance.
